src/ui/states/state-needs-auth-info.svelte

```diff
@@ -12,22 +12,39 @@
   import Localized from "../localization/localized.svelte";
   import { translatorContextKey } from "../localization/constants";
   import { Translator } from "../localization/translator";
+  import {
+    appUserIdContextKey,
+    eventsTrackerContextKey,
+    userIsAnonymousContextKey,
+  } from "../constants";
 
   import { LocalizationKeys } from "../localization/supportedLanguages";
+  import { IEventsTracker } from "../../behavioural-events/events-tracker";
 
   export let onContinue: any;
   export let onClose: () => void;
   export let processing: boolean;
   export let lastError: PurchaseFlowError | null;
 
+  const eventsTracker: IEventsTracker = getContext(eventsTrackerContextKey);
+  const appUserId: string = getContext(appUserIdContextKey);
+  const userIsAnonymous: boolean = getContext(userIsAnonymousContextKey);
+
   $: email = "";
   $: error = "";
   $: inputClass = error ? "error" : "";
 
   const handleContinue = async () => {
-    const verificationErrors = validateEmail(email);
-    if (verificationErrors) {
-      error = verificationErrors;
+    const emailError = validateEmail(email);
+    if (emailError) {
+      eventsTracker.trackBillingEmailEntryError({
+        appUserId: appUserId,
+        userIsAnonymous,
+        errorCode: 0,
```

Comment:
This needs to be fixed. The local handling of errors does not map to a `PurchaseFlowErrorCode`. Perhaps these error should only have a message and no error code? Thoughts?

Reply:
Hmm maybe we could expose the backend provided error directly and send that?